smoke-test/k0sctl.yaml.tpl

```diff
@@ -9,6 +9,12 @@ spec:
         address: "127.0.0.1"
         port: 9022
         keyPath: ./id_rsa_k0s
+      hooks:
+        apply:
+          before:
+            - "date > /before-apply.hook"
+          after:
+            - "date > /after-apply.hook"
```

Comment:
you can actually make this a sort of a test by doing something like:

```yaml
before:
  - echo hello > apply.hook
after:
  - grep -q hello apply.hook
```

this will fail the smoke if the file doesn't get the "hello" content.

Reply:
(it will not fail though if the hooks thing doesn't work at all)

Reply:
Added a separate "external" check to look into the smoke container to verify the hook has been run